source/lmp/pair_deepmd.cpp

```diff
@@ -783,6 +783,17 @@ void PairDeepMD::compute(int eflag, int vflag) {
           // Gather std_f and tags
           tagint *tag = atom->tag;
           int nprocs = comm->nprocs;
+          int newntotal = atom->natoms;
+          if (newntotal > sizeof(stdfsend)) {
```

Comment:
i am afraid the `sizeof(stdfsend)` is not what you want, see the example below:
```c++
  double aa[10];                                                                
  std::cout << sizeof(aa) << std::endl;  
```
prints
```shell
80
```

Reply:
As there is no way to get size of a pointer to the array, I just defined a new private parameter `max_nlocal` for `PairDeepMD`. Set default to 0, it would be set to `ntotal` in `init_style` and growing larger when `nlocal` (set to `atom->nloc`) becomes larger than it, to decide whether recreate `stdfsend` etc.